test/functional/rpc_blockchain.py

```diff
@@ -352,6 +355,42 @@ def assert_waitforheight(height, timeout=2):
         assert_waitforheight(current_height)
         assert_waitforheight(current_height + 1)
 
+    def _test_getblock(self):
+        if not self.is_wallet_compiled():
+            self.log.info("Skipping getblock tests because wallet is disabled")
+            return
+
+        node = self.nodes[0]
+        address = node.getnewaddress()
+        node.sendtoaddress(address, 1)
+        blockhash = node.generatetoaddress(1, address)[0]
+
+        self.log.info("Test that getblock with verbosity 1 doesn't include fee")
+        block = node.getblock(blockhash, 1)
+        assert 'fee' not in block['tx'][1]
+
+        self.log.info('Test that getblock with verbosity 2 includes expected fee')
+        block = node.getblock(blockhash, 2)
+        assert 'fee' in block['tx'][1]
+        assert_equal(block['tx'][1]['fee'], Decimal('0.00004380'))
```

Comment:
Unless you're setting amounts/fee explicitly, this feels brittle (i.e. this test would break if an unrelated change was made in how the wallet code sets fees).